app/helpers/local_time_helper.rb

```diff
@@ -42,20 +48,38 @@ def utc_time(time_or_date)
   end
 
   private
+
     def find_time_format(format)
+      find_format(format, find_i18n_time_format(format), find_ruby_time_format(format), LocalTime.default_time_format)
+    end
+
+    def find_date_format(format)
+      find_format(format, find_i18n_date_format(format), find_ruby_date_format(format), LocalTime.default_date_format)
+    end
+
+    def find_format(format, i18n_format, ruby_format, default_format)
       if format.is_a?(Symbol)
-        if (i18n_format = I18n.t("time.formats.#{format}", default: [:"date.formats.#{format}", ''])).present?
+        if i18n_format
           i18n_format
-        elsif (date_format = Time::DATE_FORMATS[format] || Date::DATE_FORMATS[format])
-          date_format.is_a?(Proc) ? LocalTime.default_time_format : date_format
+        elsif ruby_format
+          ruby_format.is_a?(Proc) ? default_format : ruby_format
         else
-          LocalTime.default_time_format
+          default_format
         end
       else
-        format.presence || LocalTime.default_time_format
+        format.presence || default_format
       end
     end
 
+    def find_i18n_time_format(format); find_i18n_format(format, :time); end
+    def find_i18n_date_format(format); find_i18n_format(format, :date); end
+    def find_i18n_format(format, type)
+      I18n.t("#{type}.formats.#{format}", default: "").presence
+    end
+
+    def find_ruby_time_format(format); Time::DATE_FORMATS[format]; end
+    def find_ruby_date_format(format); Date::DATE_FORMATS[format]; end
```

Comment:
Now that we've pulled the ruby format lookup into its own method, I think the discernment can go there. Maybe something like:

```ruby
def ruby_time_or_date_format(name, prefer: :time)
  if prefer == :time
    Time::DATE_FORMATS.with_indifferent_access[name] || Date::DATE_FORMATS.with_indifferent_access[name]
  else
    Date::DATE_FORMATS.with_indifferent_access[name] || Time::DATE_FORMATS.with_indifferent_access[name]
  end
end
```

Same goes for i18n lookup.

Feel free to go with something else if it makes sense, though.

Reply:
Also, the name `ruby_time_or_date_format` was chosen to reflect that `time` has preference over `date` – thus coming first in the method name. This would now be communicated via the named param. So renaming might be a good idea. Just make sure to pick something that won't clash with other helpers in people's apps.